_blog/2018/egress-tcp.md

```diff
@@ -0,0 +1,306 @@
+
+3. I inspect the created ratings to see that everything worked as expected:
+   ```bash
+   mysqlsh --sql --ssl-mode=REQUIRED -u bookinfo -p --host <the database host> --port <the database port> \
+   -e "select * from test.ratings;"
+   ```
+   ```bash
+   Enter password:
+   +----------+--------+
+   | ReviewID | Rating |
+   +----------+--------+
+   |        1 |      5 |
+   |        2 |      4 |
+   +----------+--------+
+   ```
+
+   _**OR**_
+
+   For `mysql` and the local database:
+   ```bash
+   mysql -u bookinfo -p -e "select * from test.ratings;"
+   ```
+   ```bash
+   Enter password:
+   +----------+--------+
+   | ReviewID | Rating |
+   +----------+--------+
+   |        1 |      5 |
+   |        2 |      4 |
+   +----------+--------+
+   ```
+
+4. I set the ratings temporarily to 1 to provide a visual clue when our database is used by the Bookinfo _ratings_ service:
+   ```bash
+   mysqlsh --sql --ssl-mode=REQUIRED -u admin -p --host <the database host> --port <the database port>  \
+   -e "update test.ratings set rating=1; select * from test.ratings;"
+   ```
+   ```bash
+   Enter password:
+   +----------+--------+
+   | ReviewID | Rating |
+   +----------+--------+
+   |        1 |      1 |
+   |        2 |      1 |
+   +----------+--------+
+   ```
+
+   _**OR**_
+
+   For `mysql` and the local database:
+   ```bash
+   mysql -u root -p -e "update test.ratings set rating=1; select * from  test.ratings;"
+   ```
+   ```bash
+   Enter password:
+   +----------+--------+
+   | ReviewID | Rating |
+   +----------+--------+
+   |        1 |      1 |
+   |        2 |      1 |
+   +----------+--------+
+   ```
+   I used the _admin_ user (and _root_ for the local database) in the last command since the _bookinfo_ user does not have the _UPDATE_ privilege on the `test.ratings` table.
+
+Now I am ready to deploy a version of the Bookinfo application that will use my database.
+
+### Initial setting of Bookinfo application
+To demonstrate the scenario of using an external database, I start with a Kubernetes cluster with [Istio installed]({{home}}/docs/setup/kubernetes/quick-start.html#installation-steps). Then I deploy [Istio Bookinfo sample application]({{home}}/docs/guides/bookinfo.html). This application uses the _ratings_ microservice to fetch book ratings, a number between 1 and 5. The ratings are displayed as stars per each review. There are several versions of the _ratings_ microservice. Some use [MongoDB](https://www.mongodb.com), others use [MySQL](https://www.mysql.com) as their database.
+
+The example commands in this blog post work with Istio version 0.3+, with or without [Mutual TLS]({{home}}/docs/concepts/security/mutual-tls.html) enabled.
+
+Let me remind you the end-to-end architecture of the application from the original [Bookinfo Guide]({{home}}/docs/guides/bookinfo.html).
```

Comment:
Let me remind you of the end-to-end ...

or maybe better:

As a reminder, here is the end-to-end architecture of the application from the [Bookinfo Guide]({{home}}/docs/guides/bookinfo.html).